File: c/003.c
#include <math.h>
#include <stdbool.h>
#include <stdio.h>

int is_even(int n)
{
	return n % 2 == 0;
}

int is_odd(int n)
{
	return n % 2 != 0;
}

int is_prime(int n)
{
	if (is_even(n))
		return 0;

	int numbers[n];

	int i;
	for (i = 0; i < n; n++) {
		if (is_odd(n)) {
			// do cool stuff
			// lol
		}
	}

	return 0;
}

int main(void)
{
	return 0;
}

File: python/003.py

import math

def is_prime(n):
	if n % 2 == 0 and n > 2: 
		return False
	return all(n % i for i in range(3, int(math.sqrt(n)) + 1, 2))

N = 600851475143

for i in range(N - 2, 0, -2):
	if (N / i).is_integer() and is_prime(i):
		break

print(i)

